pkg/k8shandler/cluster.go

```diff
@@ -200,17 +201,22 @@ func (er *ElasticsearchRequest) getNodeUpgradeInProgress() NodeTypeInterface {
 
 func (er *ElasticsearchRequest) progressUnschedulableNodes() error {
 	cluster := er.cluster
+	clusterNodes := nodes[nodeMapKey(cluster.GetName(), cluster.GetNamespace())]
+
+	for _, nodeStatus := range cluster.Status.Nodes {
+		if isPodUnschedulableConditionTrue(nodeStatus.Conditions) ||
+			isPodImagePullBackOff(nodeStatus.Conditions) ||
+			isPodCrashLoopBackOff(nodeStatus.Conditions) {
+
+			for _, node := range clusterNodes {
+				if nodeStatus.DeploymentName == node.name() || nodeStatus.StatefulSetName == node.name() {
+					if node.isMissing() {
+						log.Info("Unschedulable node not existing, creating node", "node", node.name())
+						return node.create()
```

Comment:
we don't want to create here, we want to just skip trying to progress it

Reply:
As mentioned on voice and in the description above, the routines after `progressUnshedulableNodes` require conditions with `UpgradeStatus: CertRedploy` and from my smoke tests so far it is faster to recover a lost deployment here.

Reply:
i'm not following your comment about requiring conditions...
having the create in a place where it is seemingly hidden (its not obvious that we would create here) concerns me

Reply:
Based on prior off-the-record discussion, I move the back to skipping progressing upgrade nodes if the node is not existing. In addition the status update routines will prune `status.nodes` and `status.pods` for non-existing nodes first and then update the fields for existing ones.